test/test_jsx.re

```diff
@@ -0,0 +1,567 @@
+/** Ppx Tests
+
+    This file is here to torture the jsx ppx. Tests that are directly related to
+    html or svg should go to the other files.
+*/
+open Tyxml;
+
+module type LANGUAGE = {
+  include Xml_sigs.Typed_pp;
+  type wrap('a);
+  type list_wrap('a);
+  let pp_wrap:
+    ((Format.formatter, 'a) => unit, Format.formatter, wrap('a)) => unit;
+  let pp_wrap_list:
+    ((Format.formatter, 'a) => unit, Format.formatter, list_wrap('a)) => unit;
+  let totl: list_wrap(Xml.elt) => list_wrap(elt('a));
+  let toeltl: list_wrap(elt('a)) => list_wrap(Xml.elt);
+};
+
+module TyTests = (Language: LANGUAGE) => {
+  module Testable = {
+    type t = Language.list_wrap(Xml.elt);
+    let pp = (fmt, x) =>
+      Language.pp_wrap_list(Language.pp_elt(), fmt, Language.totl(x));
+    let equal = (==);
+  };
+
+  let make = l => {
+    let f = ((name, ty1, ty2)) => (
+      name,
+      `Quick,
+      () =>
+        Alcotest.(check((module Testable)))(
+          name,
+          Language.toeltl(ty1),
+          Language.toeltl(ty2),
+        ),
+    );
+
+    List.map(f, l);
+  };
+};
+
+module Html = {
+  include Tyxml.Html;
+  let pp_wrap = pp => pp;
+  let pp_wrap_list = pp => Format.pp_print_list(~pp_sep=(_, ()) => (), pp);
+};
+module Svg = {
+  include Tyxml.Svg;
+  let pp_wrap = pp => pp;
+  let pp_wrap_list = pp => Format.pp_print_list(~pp_sep=(_, ()) => (), pp);
+};
+module HtmlTests = TyTests(Html);
+module SvgTests = TyTests(Svg);
+
+let basics = (
+  "ppx basics",
+  HtmlTests.make(
+    Html.[
+      ("elems", [[<p />]], [p([])]),
+      (
+        "child",
+        [[<p> <span> foo </span> </p>]],
+        [p([span([txt("foo")])])],
+      ),
+      (
+        "list",
+        [<> <p /> <span> foo </span> </>],
+        [p([]), span([txt("foo")])],
+      ),
+      ("attrib", [[<p id=foo />]], [p(~a=[a_id("foo")], [])]),
+      (
+        "attribs",
+        [[<p id=foo className=bar />]],
+        [p(~a=[a_id("foo"), a_class(["bar"])], [])],
+      ),
+      /* can this be done in JSX?
+         ("comment", [[%html "<!--foo-->"]], [tot @@ Xml.comment("foo")]),
+         */
+      ("txt", [[foo]], [txt("foo")]),
+      (
+        "document",
+        [[<html> <head> <title> foo </title> </head> <body /> </html>]],
+        [html(head(title(txt("foo")), []), body([]))],
+      ),
+      (
+        "let",
+        [
+          {
+            let x = <p />;
+            x;
+          },
+        ],
+        [p([])],
+      ),
+      (
+        "nested let",
+        [
+          {
+            let _ = <p />;
+            let y = <p />;
+            y;
+          },
+        ],
+        [p([])],
+      ),
+      (
+        "let and",
+        {
+          let x = <p />
+          and y = <a />;
+          [x, y];
+        },
+        [p([]), a([])],
+      ),
+      (
+        "let fun",
+        [
+          {
+            let%html f = x => "<p>"(x, "</p>");
```

Comment:
this one was not translated?

Reply:
That's fixed, had 1 more that I missed as well